src/Endatix.Core/UseCases/FormDefinitions/PartialUpdateActive/PartialUpdateActiveFormDefinitionHandler.cs

```diff
@@ -18,10 +19,24 @@ public async Task<Result<FormDefinition>> Handle(PartialUpdateActiveFormDefiniti
             return Result.NotFound("Active form definition not found.");
         }
 
-        activeDefinition.UpdateSchema(request.JsonData);
-        activeDefinition.UpdateDraftStatus(request.IsDraft);
+        if(await FormDefinitionHasSubmissions(activeDefinition.Id) && activeDefinition.JsonData != request.JsonData) {
+            var newFormDefinition = new FormDefinition(jsonData: request.JsonData);
+            await formRepository.AddNewFormDefinitionAsync(formWithActiveDefinition, newFormDefinition, cancellationToken);
```

Comment:
I am wondering if we need this method at all? Can we just add the FormDefinition using the Form entity, set it as Active and then call the `await formRepository.UpdateAsync(formWithActiveDefinition, cancellationToken);` e.g. don't do this SaveChanges?

Will be great if it works since FormDefinitions are tracked by the parent Form entity. On the dark side, there might be some EF exceptions.... You don't have to do anything before the merge, but wondering if you have tried 

In both cases, I would like us to try this when we have a bit more time and see if we can handle the whole thing in one SaveChanges call. 

P.S. I have checked the `CreateFormDefinitionHandler` and it looks it will work in which case you can delete this repository method and just go with this approach using the Form as your aggregate root and update the FormDefinition through it:

```c#
var newFormDefinition = new FormDefinition(request.IsDraft, request.JsonData);
        form.AddFormDefinition(newFormDefinition);
        if (!newFormDefinition.IsDraft)
        {
            form.SetActiveFormDefinition(newFormDefinition);
        }
        await formsRepository.UpdateAsync(form, cancellationToken);
```

Reply:
I like that you have been able to remove the repository method